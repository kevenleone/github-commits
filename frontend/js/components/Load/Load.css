.Load {
  position: fixed;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
  display: none;
  opacity: 0;
  transition-duration: 100ms;
  transition-property: all;
  display: block;
  opacity: 1;
  z-index: 99999;
}

.Load .background {
  position: fixed;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
  background: #343a40; /* Old browsers */
  background: -webkit-linear-gradient(top, #343a40 68%, #343a40 100%); /* Chrome10-25,Safari5.1-6 */
}

.Load .image {
  top: 0px;
  left: 0px;
  right: 0px;
  position: fixed;
}

.Load .image .lds-css {
  margin: calc(50vh - 100px) calc(50vw - 50px) 0px;
}

.Load h5 {
  color: #fff;
}

@keyframes lds-rolling {
  0% {
    -webkit-transform: translate(-50%, -50%) rotate(0deg);
    transform: translate(-50%, -50%) rotate(0deg);
  }
  100% {
    -webkit-transform: translate(-50%, -50%) rotate(360deg);
    transform: translate(-50%, -50%) rotate(360deg);
  }
}
@-webkit-keyframes lds-rolling {
  0% {
    -webkit-transform: translate(-50%, -50%) rotate(0deg);
    transform: translate(-50%, -50%) rotate(0deg);
  }
  100% {
    -webkit-transform: translate(-50%, -50%) rotate(360deg);
    transform: translate(-50%, -50%) rotate(360deg);
  }
}

.LoadImage .lds-rolling div,
.LoadImage .lds-rolling div:after {
  position: absolute;
  width: 50px;
  height: 50px;
  border: 4px solid #dbdbdb;
  border-top-color: #ffffff;
  border-radius: 50%;
}

.LoadImage .lds-rolling div {
  -webkit-animation: lds-rolling 3.4s linear infinite;
  animation: lds-rolling 1s linear infinite;
  top: 50px;
  left: 50px;
}

.LoadImage .lds-rolling div:after {
  -webkit-transform: rotate(90deg);
  transform: rotate(90deg);
}

.LoadImage .lds-rolling {
  width: 100px !important;
  height: 100px !important;
  -webkit-transform: translate(-50px, -50px) scale(1) translate(50px, 50px);
  transform: translate(-50px, -50px) scale(1) translate(50px, 50px);
}

.LoadImage .lds-rolling div,
.LoadImage .lds-rolling div:after {
  position: absolute;
  width: 50px;
  height: 50px;
  border: 4px solid #ececec;
  border-top-color: #baa7f9;
  border-radius: 50%;
}
