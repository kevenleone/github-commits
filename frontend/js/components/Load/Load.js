import React from 'react';
import { useSelector } from 'react-redux';
import { When } from 'react-if';

import './Load.css';

export default function Load() {
  const { loading } = useSelector((state) => state.base);
  return (
    <When condition={loading}>
      <div className="Load">
        <div className="background" />
        <div className="image">
          <div className="LoadImage lds-css">
            <div className="lds-rolling">
              <div />
            </div>
          </div>
          <h5 className="text-center pl-3 pr-3">Loading</h5>
        </div>
      </div>
    </When>
  );
}
