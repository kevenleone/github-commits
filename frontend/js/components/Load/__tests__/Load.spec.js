import React from 'react';
import { Provider } from 'react-redux';
import sagaMiddleware from 'redux-saga';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import Load from '../Load';

const mockStore = configureMockStore([sagaMiddleware]);

describe('Load', () => {
  test('should render with success', () => {
    const store = mockStore({
      base: { loading: true },
    });
    const wrapper = mount(
      <Provider store={store}>
        <Load />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
