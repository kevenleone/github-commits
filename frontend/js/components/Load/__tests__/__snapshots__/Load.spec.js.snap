// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load should render with success 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Load>
    <When
      condition={true}
    >
      <div
        className="Load"
      >
        <div
          className="background"
        />
        <div
          className="image"
        >
          <div
            className="LoadImage lds-css"
          >
            <div
              className="lds-rolling"
            >
              <div />
            </div>
          </div>
          <h5
            className="text-center pl-3 pr-3"
          >
            Loading
          </h5>
        </div>
      </div>
    </When>
  </Load>
</Provider>
`;
